### PR TITLE
Weld 2093 intercept default methods

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Comment by tremes** [comment]
I think we need tu update WildFly to use jboss-classfilewriter 1.2.0.Beta1

File: impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java [hunk 1]
```diff
@@ -493,12 +505,22 @@ protected void addAdditionalInterfaces(Set<Class<?>> interfaces) {
         return proxyClass;
     }
 
-    private static ClassFile newClassFile(String name, int accessFlags, String superclass, String... interfaces) {
-        if (CONFIGURABLE_ACCESS_FLAGS) {
-            return new ClassFile(name, accessFlags, superclass, interfaces);
-        } else {
-            return new ClassFile(name, superclass, interfaces);
+    private ClassFile newClassFile(String name, int accessFlags, String superclass, String... interfaces) {
+        WeldClassLoaderResourceLoader resourceLoader = WeldClassLoaderResourceLoader.INSTANCE;
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java]: We should probably use `this.classLoader`

**Comment by mkouba** [comment]
For now we can mark the test with `@Category(EmbeddedContainer.class)`.

**Comment by tremes** [comment]
Closed in favor of https://github.com/weld/core/pull/1415

**Pull request closed without merging**